commit a0e2516e0f10619ef708109ff3bc6b54b44d4193
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Fri Apr 4 20:39:08 2014 +0200

    Use only clock_gettime, do not depend on librt
    
    clock_gettime is actually provided by libc, not librt, so we don't need
    the latter, and thus avoid the librt dependency, which would cause
    initialization issues.
    
    * sysdeps/generic/pt-condattr-setclock.c (pthread_condattr_setclock):
    Use clock_gettime instead of clock_getres to test for clock
    availability.
    * Makefile: Do not link against librt.

diff --git a/libpthread/Makefile b/libpthread/Makefile
index 8ef990e..2f45c1c 100644
--- a/libpthread/Makefile
+++ b/libpthread/Makefile
@@ -216,10 +216,8 @@ VPATH += $(SYSDEP_PATH)
 
 ifeq ($(IN_GLIBC),no)
 HURDLIBS = ihash
-LDLIBS = -lrt
 else
 LDLIBS-pthread.so = -lihash
-$(objpfx)libpthread.so: $(common-objpfx)rt/librt.so
 endif
 
 ifeq ($(IN_GLIBC),no)
diff --git a/libpthread/sysdeps/generic/pt-condattr-setclock.c b/sysdeps/generic/pt-condattr-setclock.c
index c5a78ef..c8ac4ce 100644
--- a/libpthread/sysdeps/generic/pt-condattr-setclock.c
+++ b/libpthread/sysdeps/generic/pt-condattr-setclock.c
@@ -35,7 +35,7 @@ pthread_condattr_setclock (pthread_condattr_t *attr, clockid_t clock)
 	  struct timespec ts;
 	  int res;
 
-	  res = clock_getres (CLOCK_MONOTONIC, &ts);
+	  res = clock_gettime (CLOCK_MONOTONIC, &ts);
 	  avail = res < 0 ? -1 : 1;
 	}
 
